packages/fiori-elements-writer/src/index.ts

```diff
@@ -101,6 +101,10 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
     // Define template options with changes preview and loader settings based on project type
     const templateOptions = getTemplateOptions(isEdmxProjectType, feApp.service.version, feApp.ui5?.version);
 
+    // FPM to preload sap.fe.templates for FLP template
+    if (feApp.template.type === TemplateType.FlexibleProgrammingModel) {
+        (ui5Libs as string[]).push('sap.fe.templates');
```

Comment:
Shouldnt ui5Libs be undefined for CAP (as per https://github.com/SAP/open-ux-tools/pull/2237/files#diff-a51a19d6fa0b94939e84e2a4eedd1b602cae40c2077c22ecce59ee537f5ccbcfR100) . If so I would suggest combining line: 100 and line 105-107 into a single function for clarity?

Reply:
addressed in https://github.com/SAP/open-ux-tools/pull/2237/commits/5b6e9f5ff7115d9558424cd2d49003b84500bb67
Thanks